Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/CruiseControl.cs

```diff
@@ -0,0 +1,2277 @@
+            ForceResetRequiredAfterBraking = other.ForceResetRequiredAfterBraking;
+            ForceResetIncludeDynamicBrake = other.ForceResetIncludeDynamicBrake;
+            ZeroSelectedSpeedWhenPassingToThrottleMode = other.ZeroSelectedSpeedWhenPassingToThrottleMode;
+            DynamicBrakeCommandHasPriorityOverCruiseControl = other.DynamicBrakeCommandHasPriorityOverCruiseControl;
+            HasIndependentThrottleDynamicBrakeLever = other.HasIndependentThrottleDynamicBrakeLever;
+            HasProportionalSpeedSelector = other.HasProportionalSpeedSelector;
+            DisableManualSwitchToManualWhenSetForceNotAtZero = other.DisableManualSwitchToManualWhenSetForceNotAtZero;
+            DisableManualSwitchToAutoWhenThrottleNotAtZero = other.DisableManualSwitchToAutoWhenThrottleNotAtZero;
+            DisableManualSwitchToAutoWhenSetSpeedNotAtTop = other.DisableManualSwitchToAutoWhenSetSpeedNotAtTop;
+            EnableSelectedSpeedSelectionWhenManualModeSet = other.EnableSelectedSpeedSelectionWhenManualModeSet;
+            SpeedSelectorIsDiscrete = other.SpeedSelectorIsDiscrete;
+            DoComputeNumberOfAxles = other.DoComputeNumberOfAxles;
+            UseTrainBrakeAndDynBrake = other.UseTrainBrakeAndDynBrake;
+            SpeedDeltaToEnableTrainBrake = other.SpeedDeltaToEnableTrainBrake;
+            SpeedDeltaToEnableFullTrainBrake = other.SpeedDeltaToEnableFullTrainBrake;
+            MinimumSpeedForCCEffectMpS = other.MinimumSpeedForCCEffectMpS;
+            TrainBrakeMinPercentValue = other.TrainBrakeMinPercentValue;
+            TrainBrakeMaxPercentValue = other.TrainBrakeMaxPercentValue;
+            StartInAutoMode = other.StartInAutoMode;
+            ThrottleNeutralPosition = other.ThrottleNeutralPosition;
+            ThrottleLowSpeedPosition = other.ThrottleLowSpeedPosition;
+            LowSpeed = other.LowSpeed;
+            HasTwoForceValues = other.HasTwoForceValues;
+
+        }
+
+        public void Initialize()
+        {
+            Simulator = Locomotive.Simulator;
+            clockTime = Simulator.ClockTime * 100;
+            ComputeNumberOfAxles();
+            if (StartReducingSpeedDeltaDownwards == 0) StartReducingSpeedDeltaDownwards = StartReducingSpeedDelta;
+            if (StartInAutoMode) SpeedRegMode = SpeedRegulatorMode.Auto;
+        }
+
+        private void ComputeNumberOfAxles()
+        {
+            if (DoComputeNumberOfAxles && (TrainCar)Locomotive == Simulator.PlayerLocomotive)
+            {
+                SelectedNumberOfAxles = 0;
+                foreach (TrainCar tc in Locomotive.Train.Cars)
+                {
+                    SelectedNumberOfAxles += tc.WheelAxles.Count;
+                }
+            }
+        }
+
+        public void Update(float elapsedClockSeconds)
+        {
+            OverrideForceCalculation = false;
+            if (!Locomotive.IsPlayerTrain)
+            {
+                WasForceReset = false;
+                controllerVolts = 0;
```

Comment:
This is the key variable which governs the Cruise Control. It determines the percentage of throttle and brake applied. I think it deserves a more descriptive name.

Reply:
Done. Added also a comment explaining what the variable does.